media/js/firefox/accounts.js

```diff
@@ -20,6 +24,18 @@
         fxaIframeSrc = fxaIframeSrc.replace('context=iframe', 'context=fx_firstrun_v2');
     }
 
+    // check for fxaEmail in sessionStorage or URL params
+    try {
+        fxaEmail = sessionStorage.getItem('fxa-email');
+    } catch(ex) {
+        fxaEmail = params.get('fxa-email');
```

Comment:
It makes me a little nervous that we're potentially passing user email addresses in the URL here. Under what circumstances does this happen, only if `sessionStorage` fails? Given that this page is a public URL which is also soon to be indexed by search engines, even if someone shares the URL on twitter without realizing their email is in the URL could be bad?

Do we also need to think about validating the email address, to try and prevent any potential XSS vectors?